lib/wallet/account.js

```diff
@@ -386,6 +386,45 @@ class Account {
     return key;
   }
 
+  /**
+   * Derive an address (doesn't increment depth).
+   * @param {Number} branch
+   * @param {Number} index
+   * @returns {Promise} - Returns {@link WalletKey}.
+   */
+
+  createKeyAtIndex(branch, index, force) {
```

Comment:
How is this function different from `deriveKey(branch, index)`?

Reply:
`createKeyAtIndex` is a wrapper around `deriveKey` (it calls `deriveReceive`, `deriveChange` or `deriveNested` based on `branch`, and those three functions in turn call `deriveKey`).

I think the main reason I wrote that as a separate function is to check the requested index depth. Because you can derive a key at any index, a user could generate an address that is not yet being watched by the wallet (i.e. the wallet depth is only `20` but user derives key at index `100` and sends bitcoin to that address -- the wallet would not be looking for that transaction) There is a `force` flag to bypass this check.